src/functionlenses.jl

```diff
@@ -31,6 +31,17 @@ function set(x::TX, f::Base.Fix1{typeof(convert)}, v) where {TX}
     convert(TX, v)
 end
 
+set(obj::Tuple, ::Type{Tuple}, val::Tuple) = val
+set(obj::NamedTuple{KS}, ::Type{Tuple}, val::Tuple) where {KS} = NamedTuple{KS}(val)
+set(obj::CartesianIndex, ::Type{Tuple}, val::Tuple) = CartesianIndex(val)
+set(obj::AbstractVector, ::Type{Tuple}, val::Tuple) = similar(obj, eltype(val)) .= val
```

Comment:
Interesting, so this would update an `SVector` into an `MVector`, which is maybe unexpected?